### PR TITLE
mrb: enable mruby-dir for Dir#foreach

### PR DESCRIPTION
**Pull request by kenhys**
grndb --force-truncate use Dir#foreach so, this PR enable mruby-dir for it.

### CODE REVIEW

**Comment by kenhys** [comment]
windowsでビルドできない問題ではまっている。

**Comment by kenhys** [comment]
```
  CXXLD    libgroonga.la
../vendor/mruby/.libs/libmruby.a(dirent.o): In function `opendir':
/tmp/groonga/groonga-7.0.4/vendor/mruby/mruby-dir/src/Win/dirent.c:39: multiple definition of `opendir'
../vendor/mruby/.libs/libmruby.a(dir.o):/tmp/groonga/groonga-7.0.4/vendor/mruby/mruby-dir/src/Win/dirent.c:39: first defined here
../vendor/mruby/.libs/libmruby.a(dirent.o): In function `closedir':
/tmp/groonga/groonga-7.0.4/vendor/mruby/mruby-dir/src/Win/dirent.c:81: multiple definition of `closedir'
../vendor/mruby/.libs/libmruby.a(dir.o):/tmp/groonga/groonga-7.0.4/vendor/mruby/mruby-dir/src/Win/dirent.c:81: first defined here
../vendor/mruby/.libs/libmruby.a(dirent.o): In function `readdir':
/tmp/groonga/groonga-7.0.4/vendor/mruby/mruby-dir/src/Win/dirent.c:104: multiple definition of `readdir'
../vendor/mruby/.libs/libmruby.a(dir.o):/tmp/groonga/groonga-7.0.4/vendor/mruby/mruby-dir/src/Win/dirent.c:104: first defined here
../vendor/mruby/.libs/libmruby.a(dirent.o): In function `rewinddir':
/tmp/groonga/groonga-7.0.4/vendor/mruby/mruby-dir/src/Win/dirent.c:124: multiple definition of `rewinddir'
../vendor/mruby/.libs/libmruby.a(dir.o):/tmp/groonga/groonga-7.0.4/vendor/mruby/mruby-dir/src/Win/dirent.c:124: first defined here
collect2: error: ld returned 1 exit status
```

File: vendor/mruby/Makefile.am [hunk 1]
```diff
@@ -37,6 +37,9 @@ endif
 
 include sources.am
 include built_sources.am
+if PLATFORM_WIN32
+BUILT_SOURCES += mruby-dir/src/Win/dirent.c
```
Comment (kou) [inline vendor/mruby/Makefile.am]: `src/Win/dirent.c` is included in `src/dir.c`: https://github.com/iij/mruby-dir/blob/master/src/dir.c#L19
So we don't need to build it manually.

We need to just bundle it into our tar.gz.

Comment (kou) [inline vendor/mruby/Makefile.am]: We can use `EXTRA_DIST` in `Makefile.am`.

Comment (kenhys) [inline vendor/mruby/Makefile.am]: Thanks, I'll try it.

**Comment by kenhys** [comment]
I've confirmed on windows.

**Pull request merged**